src/app/static/src/app/utils.ts

```diff
@@ -37,33 +38,44 @@ export const mapStatePropByName = <T>(namespace: string | null, name: string): C
     return (namespace && mapState(namespace, [name])[name]) || mapState([name])[name]
 };
 
-export const mapStateProps = <S, K extends string>(namespace: string,
-    map: Dict<(this: CustomVue, state: S) => any>) => {
-    type R = {[key in K]: any}
-    return mapState<S>(namespace, map) as R
+export const mapStateProps = <M extends Dict<(this: CustomVue, state: any) => any>>(namespace: string, map: M) => {
+    type T = typeof map
+    type R = {[key in keyof T]: ComputedGetter<ReturnType<T[key]>>}
+    return mapState(namespace, map) as R
 };
 
-export const mapGetterByName = <T>(namespace: string | null, name: string): ComputedWithType<T> => {
+export const mapRootStateProps = <M extends Dict<(this: CustomVue, state: any) => any>>(map: M) => {
+    type T = typeof map
+    type R = {[key in keyof T]: ComputedGetter<ReturnType<T[key]>>}
+    return mapState(map) as R
+};
+
+export const mapGetterByName = <T>(namespace: string | null, name: string): ComputedGetter<T> => {
     return (namespace && mapGetters(namespace, [name])[name]) || mapGetters([name])[name]
 }
 
-export const mapGettersByNames = <K extends string>(namespace: string, names: string[]) => {
-    type R = {[key in K]: any}
-    return mapGetters(namespace, names) as R
+export const mapGettersByNames = <N extends readonly string[], T extends Record<N[number], any>>(namespace: string, names: N) => {
+    const nameCopy = [...names]
+    type R = {[key in N[number]]: any}
+    type TR = {[key in N[number]]: ComputedGetter<T[key]>}
+    type Result = T extends undefined ? R : TR
+    return mapGetters(namespace, nameCopy) as Result
 };
```

Comment:
Can you include a return Type ?

Can you also allow namespace to accept null value ?

Reply:
return type is Result, see earlier comment about types, will leave namespace as it is only because hint doesn't use null anywhere and the main purpose of the PR is to coordinate types for defineComponent and Vue instead of changing what we allow users to input, can create a ticket for it, but will involve thinking a little deeper about this to see how null plays into the types